hardemu/screen.js

```diff
@@ -1189,19 +1197,64 @@ function VideoChip(Zeal, PIO, scale) {
             vConfigWrite(port, value);
         } else if (port >= 0x10 && port <= 0x1f) {
             // TODO: Scrolling values
-            if (port == 0x1c)
-                updateModeData(canvas, canvas_layer1, value);
-            else if (port == 0x1d)
-                video_cfg.screen_enabled = (value >> 7) & 1;
+            switch(port) {
+                case CTRL_BASE.L0_SCR_Y_LOW: // zvb_ctrl_l0_scr_y_low\
+                    gfx_cfg.scroll_l0_y_latch = value;
+                    // gfx_cfg.scroll_l0_y &= 0xFF00;
+                    // gfx_cfg.scroll_l0_y += value;
+                    break;
+                case CTRL_BASE.L0_SCR_Y_HIGH: // zvb_ctrl_l0_scr_y_high
+                    // gfx_cfg.scroll_l0_y &= 0x00FF;
+                    gfx_cfg.scroll_l0_y = (value << 8) + gfx_cfg.scroll_l0_y_latch;
+                    break;
+                case CTRL_BASE.L0_SCR_X_LOW: // zvb_ctrl_l0_scr_x_low
+                    gfx_cfg.scroll_l0_x_latch = value;
+                    // gfx_cfg.scroll_l0_x &= 0xFF00;
+                    // gfx_cfg.scroll_l0_x += value;
+                    break;
+                case CTRL_BASE.L0_SCR_X_HIGH: // zvb_ctrl_l0_scr_x_high
+                    // gfx_cfg.scroll_l0_x &= 0x00FF;
+                    gfx_cfg.scroll_l0_x = (value << 8) + gfx_cfg.scroll_l0_x_latch;
+                    break;
+
+                case CTRL_BASE.L1_SCR_Y_LOW: // zvb_ctrl_l0_scr_y_low
+                    // gfx_cfg.scroll_l1_y &= 0xFF00;
+                    // gfx_cfg.scroll_l1_y += value;
+                    gfx_cfg.scroll_l1_y_latch = value;
+                    break;
+                case CTRL_BASE.L1_SCR_Y_HIGH: // zvb_ctrl_l0_scr_y_high
+                    // gfx_cfg.scroll_l1_y &= 0x00FF;
```

Comment:
The new implementation with the the latched values is correct, these comments can be removed